// @okta/okta-signin-widget/dist/esm/src/v2/view-builder/views/idp/RedirectIdvView.js
import _Handlebars2 from '../../../../../packages/@okta/courage-dist/esm/lib/handlebars/dist/cjs/handlebars.runtime.js';
import { View, loc } from '../../../../../packages/@okta/courage-dist/esm/src/CourageForSigninWidget.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/framework/Model.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/vendor/lib/backbone.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/util/jquery-wrapper.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/util/underscore-wrapper.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/models/Model.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/models/BaseModel.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/framework/View.js';
import '../../../../../packages/@okta/courage-dist/esm/src/courage/views/Backbone.ListView.js';
import { BaseIdPAuthenticatorBody, BaseIdpAuthenticatorView } from './BaseIdpAuthenticator.js';
import { FORMS } from '../../../ion/RemediationConstants.js';
import '../../internals/BaseHeader.js';
import BaseFooter from '../../internals/BaseFooter.js';
import '../../internals/BaseForm.js';
import '../../internals/BaseFormWithPolling.js';
import '../../internals/BaseOktaVerifyChallengeView.js';
import '../../internals/BaseView.js';
import '../../components/AuthenticatorEnrollOptions.js';
import '../../components/AuthenticatorVerifyOptions.js';
import '../../../../../packages/@okta/courage-dist/esm/lib/underscore/underscore-min.js';
import 'cross-fetch';
import '../../../../util/BrowserFeatures.js';
import '../../../../util/FactorUtil.js';
import '../../../../v1/views/admin-consent/ScopeList.js';
import '../../../../v1/views/consent/ScopeList.js';
import '../captcha/CaptchaView.js';

const PrivacyTermsFooterView = View.extend({
  tagName: 'p',
  className: 'privacy-footer',
  template: _Handlebars2.template({
    "compiler": [8, ">= 4.3.0"],
    "main": function (container, depth0, helpers, partials, data) {
      var alias1 = depth0 != null ? depth0 : container.nullContext || {},
        alias2 = container.hooks.helperMissing,
        alias3 = container.escapeExpression,
        lookupProperty = container.lookupProperty || function (parent, propertyName) {
          if (Object.prototype.hasOwnProperty.call(parent, propertyName)) {
            return parent[propertyName];
          }
          return undefined;
        };
      return "<p>" + alias3((lookupProperty(helpers, "i18n") || depth0 && lookupProperty(depth0, "i18n") || alias2).call(alias1, {
        "name": "i18n",
        "hash": {
          "$2": "<a href='https://withpersona.com/legal/privacy-policy' class='inline-link' data-se='privacy-policy' target='_blank' rel='noopener noreferrer'>$2</a>",
          "$1": "<a href='https://withpersona.com/legal/terms-of-use' class='inline-link' data-se='terms-of-use' target='_blank' rel='noopener noreferrer'>$1</a>",
          "bundle": "login",
          "code": "oie.idv.idp.description.termsOfUse"
        },
        "data": data,
        "loc": {
          "start": {
            "line": 1,
            "column": 3
          },
          "end": {
            "line": 1,
            "column": 372
          }
        }
      })) + "</p><p>" + alias3((lookupProperty(helpers, "i18n") || depth0 && lookupProperty(depth0, "i18n") || alias2).call(alias1, {
        "name": "i18n",
        "hash": {
          "bundle": "login",
          "code": "oie.idv.idp.description.agreement"
        },
        "data": data,
        "loc": {
          "start": {
            "line": 1,
            "column": 379
          },
          "end": {
            "line": 1,
            "column": 443
          }
        }
      })) + "</p>";
    },
    "useData": true
  })
});
const Body = BaseIdPAuthenticatorBody.extend({
  title: function () {
    const redirectIDVerifyRemediation = this.options.appState.get('remediations').find(remediation => {
      return remediation.name === FORMS.REDIRECT_IDVERIFY;
    });
    return loc('oie.idv.idp.title', 'login', [redirectIDVerifyRemediation.idp.name]);
  },
  subtitle: function () {
    return loc('oie.idv.idp.description', 'login');
  },
  save: function () {
    return loc('oie.optional.authenticator.button.title', 'login');
  }
});
var RedirectIdvView = BaseIdpAuthenticatorView.extend({
  Body: Body,
  Footer: BaseFooter.extend({
    postRender: function () {
      this.add(PrivacyTermsFooterView);
    }
  })
});

export { RedirectIdvView as default };
